build.psm1

```diff
@@ -3371,3 +3379,8 @@ function New-NugetConfigFile
 
     Set-Content -Path (Join-Path $Destination 'nuget.config') -Value $content -Force
 }
+
+function Test-IsArm64MacOS
+{
+    return $IsMacOS -and (Start-NativeExecution -IgnoreExitCode { (arch -arm64 uname -p) -eq 'arm' } 2>$null)
```

Comment:
Can we add this in Get-EnvironmentInfo.. that function is used by tests and packaging i believe which might need this info.